scenario/actionhandler.go

```diff
@@ -459,9 +461,70 @@ func (act *Action) endAction(sessionState *session.State, actionState *action.St
 		containerActionEntry = originalActionEntry
 	}
 
-	err := logResult(sessionState, actionState, actionState.Details, containerActionEntry)
+	if err := logResult(sessionState, actionState, actionState.Details, containerActionEntry); err != nil {
+		return err
+	}
 	sessionState.LogEntry.LogDebugf("%s END", act.Type)
-	return errors.WithStack(err)
+
+	if err := logObjectRegressionData(sessionState); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func logObjectRegressionData(sessionState *session.State) error {
+	// regression logging
+	if !sessionState.LogEntry.ShouldLogRegression() {
+		return nil
+	}
+	if sessionState.Connection == nil {
+		// some actions do not have a connetion set up
+		return nil
+	}
+	uplink := sessionState.Connection.Sense()
+	if uplink == nil {
+		return errors.New("could not log regression data: no sense connection")
+	}
+
+	// log regression analysis data for each subscribed object
+	err := uplink.Objects.ForEachWithLock(
+		func(obj *enigmahandlers.Object) error {
+			// skip sheet and app types
+			if obj.Type != enigmahandlers.ObjTypeGenericObject {
+				return nil
+			}
+
+			genType := ""
+			if genObj, ok := obj.EnigmaObject.(*enigma.GenericObject); ok {
+				genType = genObj.GenericType
+			}
+
+			hyperCube := obj.HyperCube()
+			listObject := obj.ListObject()
+
+			err := sessionState.LogEntry.LogRegression(
+				// use unique id SESSION.ACTION.OBJ
+				fmt.Sprintf("%d.%d.%s", sessionState.LogEntry.Session.Session, sessionState.Counters.ActionID.Current(), obj.ID),
```

Comment:
might want to get the actionid from log entry too, for the sake of container actions, other wise the end of the container action will have the actionID of the last subaction.

Reply:
I guess we should go with only the logEntry actionID then?

Reply:
I think that's safer bet.